hooks/shotgun_fields.py

```diff
@@ -0,0 +1,288 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook that controls how Shotgun data should be displayed in the info panel app. 
+"""
+import sgtk
+import os
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+class ShotgunConfiguration(HookBaseClass):
+        
+    def get_list_item_definition(self, entity_type):
+        """
+        Controls the rendering of items in the listings.
+        """
```

Comment:
I tinkered quite a bit by this - it was my initial approach but the config ended up being HUGE and it was really painful to maintain so I switched to a hook for the following reasons:

- There is a bunch of inheritance going on in the hook, to avoid duplication. That's harder to express in the config.
- I kept making many changes to the config throughout the dev phase. If we wanted to make changes in the future, this is much easier to do if we have a hook than if it's in the config. Similarly, its easier for people to override and tweak some things but keep most things as defaults.
- I might be wrong about this, but it seems to me that overriding this sort of thing is a slightly more advanced thing to do - it requires a template language type approach etc, so it felt ok to have it as a hook despite that being more technical.
- With a hook approach, something like the perforce framework/games cfg could redefine the config params using hook inheritance.

Hope it makes sense. Not sure it's right, i am not 100% sure about the approach, but there seemed to be enough indications pointing in the hook direction to at least go with that as an initial approach!